packages/internal-test-helpers/lib/test-cases/abstract-application.ts

```diff
@@ -1,10 +1,19 @@
-import { compile } from 'ember-template-compiler';
+import { compile, EmberPrecompileOptions } from 'ember-template-compiler';
 import { ENV } from '@ember/-internals/environment';
 import AbstractTestCase from './abstract';
 import { runDestroy, runTask, runLoopSettled } from '../run';
+import Application from '@ember/application';
+import ApplicationInstance, { BootOptions } from '@ember/application/instance';
+import Router from '@ember/routing/router';
 
-export default class AbstractApplicationTestCase extends AbstractTestCase {
-  _ensureInstance(bootOptions) {
+export default abstract class AbstractApplicationTestCase extends AbstractTestCase {
+  _applicationInstancePromise?: Promise<ApplicationInstance>;
+
+  declare application: Application;
```

Comment:
Can you add a note explaining where this comes from? `declare` in a context like this makes me twitchy. 🤪

Reply:
Probably should be abstract.